src/Video/video.tsx

```diff
@@ -0,0 +1,208 @@
+import React, { useRef, useEffect, useState } from 'react';
+import styled from 'styled-components';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import {
+  faMicrophoneSlash,
+  faMicrophone,
+  faVideo,
+  faVideoSlash,
+} from '@fortawesome/free-solid-svg-icons';
+
+const VideoContainer = styled.div`
+  position: absolute;
+  z-index: 999;
+  display: flex;
+  flex-direction: column;
+`;
+
+const Video = styled.video`
+  height: ${(props) => props.height || 150}px;
+  width: ${(props) => props.width || 150}px;
+  background-color: black;
+`;
+
+const IconContainer = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: center;
+  background-color: rgb(220, 220, 220, 0.5);
+  border-radius: 32px;
+`;
+
+const MicrophoneSlash = styled(FontAwesomeIcon)`
+  color: #ed4650;
+  margin: 2%;
+`;
+
+const Microphone = styled(FontAwesomeIcon)`
+  color: black;
+  margin: 2%;
+`;
+
+const VideoIcon = styled(FontAwesomeIcon)`
+  color: black;
+  margin: 2%;
+`;
+
+const VideoIconSlash = styled(FontAwesomeIcon)`
+  color: #ed4650;
+  margin: 2%;
+`;
+
+type VideoProps = {
+  height?: number;
+  width?: number;
+};
+
+enum VideoAudioOptions {
+  audio = 'audio',
+  video = 'video',
+  all = 'all',
+}
+
+/**
+ * VideoPlayer component renders configurable video & audio media from a user's device.
+ * Renders UI to toggle audio & video on & off.
+ *
+ * The component uses the useRef hook to obtain a reference to a <video/> component.
+ * It requests permission for audio / video with navigator.mediaDevices.getUserMedia()
+ * in useEffect hooks invoked upon component mount & upon change in audioOn or videoOn state.
+ * The srcObject of the video Ref is then set to the stream returned from the navigator browser API.
+ *
+ * On a state change update to end video or audio, endMedia is invoked with appropriate type specified.
+ * This function then stops each appropriate live track to end the stream & turn off the user's webcam light.
+ *
+ * On a state change update to start video or audio, endMedia is invoked again to clear any current active streams,
+ * then startMedia is invoked to receive a stream from the navigator browser API and reset videoRef's srcObject to this stream.
+ * @param props VideoProps
+ * @returns
+ */
+function VideoPlayer(props: VideoProps) {
+  const [audioOn, setAudioOn] = useState(true);
+  const [videoOn, setVideoOn] = useState(true);
+
+  let videoRef = useRef<HTMLVideoElement | null>(null);
+
+  function endMedia(trackKind: VideoAudioOptions) {
+    try {
+      if (videoRef.current?.srcObject) {
+        let stream = videoRef.current.srcObject as MediaStream;
+        stream.getTracks().forEach(function (track) {
+          if (trackKind === 'all') {
+            if (track.readyState == 'live') {
+              track.stop();
+            }
+          } else {
+            if (track.readyState == 'live' && track.kind === trackKind) {
+              track.stop();
+            }
+          }
```

Comment:
```suggestion
          if (trackKind === 'all' && track.readyState == 'live') {
              track.stop();
          } else if (track.readyState == 'live' && track.kind === trackKind) {
              track.stop();
          }
```

You don't need to write `if` statements nested like this, you can simplify. That being said, the second issue with the code is lack of clarity in intention. 

A general pointer -- wherever you find yourself writing complex logic, take a pause and inquire whether there is an easier or more clear way to get the job done. There often is. Once you get it done, document like hell. 

Code is not just for computers to read. It's also equally, if not more so, for humans to read! Humans are the ones that use it, modify it, refactor it, extend it. If someone with no prior context can come in and understand your code, you've done a good job (assuming it actually works as intended).
If someone with know prior knowledge